Add DisableAcceleratorUsageMetrics feature flag documentation (#22628)
kubernetes/website: content/en/docs/concepts/cluster-administration/system-metrics.md

If you're upgrading from release `1.12` to `1.13`, but still depend on a metric `A` deprecated in `1.12`, you should set hidden metrics via command line: `--show-hidden-metrics=1.12` and remember to remove this metric dependency before upgrading to `1.14`

> ## Disable Accelerator Metrics

Author: Fixed, thanks!

> Since Kubernetes release 1.9, the kubelet has been collecting Accelerator metrics through cadvisor. To collect such metrics, for accelerators such as NVIDIA GPUs, Kubelet held an open handle on that driver. This meant that to perform infrastructure changes (e.g: Update the driver), a cluster administrator needed to stop the kubelet agent.

Reviewer: ```suggestion
The kubelet collects accelerator metrics through cadvisor. To collect metrics for accelerators like NVIDIA GPUs, Kubelet held an open handle on the driver. This meant that in order to perform infrastructure changes (for example, updating the driver), a cluster administrator needed to stop the kubelet agent.
```

Author: Fixed, thanks!

> Since Kubernetes release 1.9, the kubelet has been collecting Accelerator metrics through cadvisor. To collect such metrics, for accelerators such as NVIDIA GPUs, Kubelet held an open handle on that driver. This meant that to perform infrastructure changes (e.g: Update the driver), a cluster administrator needed to stop the kubelet agent.

Reviewer 2: minor:

```suggestion
Since Kubernetes release 1.9, the kubelet has been collecting Accelerator metrics through cAdvisor. To collect these metrics for accelerators such as NVIDIA GPUs, kubelet held an open handle on that driver. This meant that to perform infrastructure changes (e.g. update the driver), a cluster administrator needed to stop the kubelet agent.
```

Author: Fixed, thanks!

Author: Fixed, thanks!

> Going forward, the responsibility for collecting Accelerator metrics has shifted towards the vendor rather than kubelet. The vendor is expected to provide a container collecting the metrics and exposing them to the metrics service (e.g: Prometheus).

Reviewer: ```suggestion
As of v1.19, the responsibility for collecting accelerator metrics belongs to the vendor rather than the kubelet. Vendors must provide a container that collects the metrics and exposes them to the metrics service (for example, Prometheus).
```

Author: I'm going to reword to not provide a reference to v1.19, the responsibility didn't shift in v1.19 :D

Author: Fixed, thanks!

> To address the above issue, the [`DisableAcceleratorUsageMetrics` feature gate](/docs/references/command-line-tools-reference/feature-gate.md) has been introduced to disable metrics collected by kubelet. Timeline for enabling this feature can be found [here](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria)

Reviewer 2: You can use a new feature to link to specific text:

```suggestion
To address the above issue, the [`DisableAcceleratorUsageMetrics` feature gate](/docs/references/command-line-tools-reference/feature-gate.md#feature-gates-for-alpha-or-beta-features:~:text= DisableAcceleratorUsageMetrics,-false) has been introduced to disable metrics collected by kubelet. Timeline for enabling this feature can be found [here](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria)
```

Author: Fixed, thanks!

> To address the above issue, the [`DisableAcceleratorUsageMetrics` feature gate](/docs/references/command-line-tools-reference/feature-gate.md) has been introduced to disable metrics collected by kubelet. Timeline for enabling this feature can be found [here](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria)

Reviewer: ```suggestion
The [`DisableAcceleratorUsageMetrics` feature gate](/docs/references/command-line-tools-reference/feature-gate.md) disables metrics collected by the kubelet, with a [timeline for enabling this feature](https://github.com/kubernetes/enhancements/tree/411e51027db842355bd489691af897afc1a41a5e/keps/sig-node/1867-disable-accelerator-usage-metrics#graduation-criteria)
```

Author: Fixed, thanks!
